inference-engine/src/mkldnn_plugin/emitters/jit_snippets_emitters.hpp

```diff
@@ -0,0 +1,573 @@
+        if (!tile->compile_params)
+            IE_THROW() << "TileEmitter invoked without compile_params";
+        _jep = *reinterpret_cast<const jit_snippets_compile_args*>(tile->compile_params);
+    }
+
+    size_t get_inputs_num() const override {return 0;}
+
+    void emit_code(const std::vector<size_t> &in, const std::vector<size_t> &out,
+              const std::vector<size_t> &pool = {}, const std::vector<size_t> &gpr = {}) const override {
+        emit_impl(in, out, pool, gpr, nullptr);
+    }
+    jit_snippets_compile_args _jep;
+
+private:
+    void emit_impl(const std::vector<size_t>& in,
+                   const std::vector<size_t>& out,
+                   const std::vector<size_t>& pool,
+                   const std::vector<size_t>& gpr,
+                   const MKLDNNPlugin::emitter_context *emit_context) const override {
+        const size_t inc = in[0];
+        const size_t prew_inc = in[1]; // increment of a prew tile in the same dim (0 if the first tile in the dim)
+        const size_t num_params = in[2];
+        const size_t dim = in[3]; // tile dimension: 0 - outer, 1 - inner
+        const int reg64_tmp_start { 8 }; // R8, R9, R10, R11, R12, R13, R14, R15 inputs+outputs+1
+        Reg64 amount = Reg64(reg64_tmp_start + num_params); // amount
+        std::array<Label, 2> for_body;
+        // If R15 is not used, reserve it for use in scalar to avoid redundant push-pop's.
+        // todo: Do we need explicitly check that code contains ScalarEmitter?
+        std::vector<size_t> local_gpr = reg64_tmp_start + num_params < 15 ? std::vector<size_t>{15} : std::vector<size_t>{};
+        std::vector<Reg64> regs(num_params);
+        for (auto i = 0; dim == 0 && i < num_params; i++)
+            regs[i] = Reg64(reg64_tmp_start + i);
+
+        if (inc > _jep.scheduler_dims[dim]) {
+            return;
+        } else if (inc == _jep.scheduler_dims[dim]) {
+            for (auto& c : code) {
+                c.first->emit_code(c.second.first, c.second.second, pool, local_gpr);
+            }
+        } else {
+            // The prew tile has done nothing, all the work is ours
+            if (prew_inc == 0 || prew_inc > _jep.scheduler_dims[dim]) {
+                h->mov(amount, _jep.scheduler_dims[dim]);
+            // The prew tile has done all the work
+            } else if (prew_inc == _jep.scheduler_dims[dim]) {
+                return;
+            }// else: the prew tile has already set a proper work amount
+            h->cmp(amount, inc);
+            h->jl(for_body[0], CodeGenerator::T_NEAR);
+
+            h->L(for_body[1]);
+            {
+                h->push(amount);
+                for (auto& c : code) {
+                    c.first->emit_code(c.second.first, c.second.second, pool, local_gpr);
+                }
+                h->pop(amount);
+                // we need to add offset for ptrs only in external tiles because stores and loaders in internal tiles have default offsets
```

Comment:
This means ptrs movement logic is covered by postIncrement inside Load/Store emitters? Maybe spread the comment?

Reply:
Done